libtenzir/src/tql2/eval_binary.cpp

```diff
@@ -191,6 +191,70 @@ struct BinOpKernel<ast::binary_op::add, duration_type, time_type> {
   }
 };
 
+template <>
+struct BinOpKernel<ast::binary_op::add, duration_type, duration_type> {
+  using result = duration;
+
+  static auto evaluate(duration l, duration r)
+    -> std::variant<result, const char*> {
+    return l + r;
+  }
+};
+
+template <>
+struct BinOpKernel<ast::binary_op::sub, duration_type, duration_type> {
+  using result = duration;
+
+  static auto evaluate(duration l, duration r)
+    -> std::variant<result, const char*> {
+    return l - r;
+  }
+};
+
+template <>
+struct BinOpKernel<ast::binary_op::div, duration_type, duration_type> {
+  using result = duration::rep;
+
+  static auto evaluate(duration l, duration r)
+    -> std::variant<result, const char*> {
+    if (r == decltype(r){}) 
+      return "division by zero";
+    return l / r;
+  }
+};
+
+template <>
+struct BinOpKernel<ast::binary_op::mul, duration_type, duration::rep> {
```

Comment:
This probably needs to be implemented for all three number types, namely `uint64`, `int64`, and `double`.